pkg/server/devfile-handler.go

```diff
@@ -0,0 +1,166 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/console/pkg/serverutils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	devfilev1 "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	devfilePkg "github.com/devfile/library/pkg/devfile"
+	"github.com/devfile/library/pkg/devfile/generator"
+	"github.com/devfile/library/pkg/devfile/parser"
+)
+
+const (
+	dockerfilePath = "Dockerfile"
+)
+
+var (
+	data       formData
+	devfileObj parser.DevfileObj
+)
+
+func (s *Server) devfileHandler(w http.ResponseWriter, r *http.Request) {
+	_ = json.NewDecoder(r.Body).Decode(&data)
+
+	var err error
+
+	// Get devfile contents and parse them using a library call in the future
+	devfileContentBytes := []byte(data.Devfile.DevfileContent)
+	devfileObj, err = devfilePkg.ParseFromDataAndValidate(devfileContentBytes)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse devfile: %v", err)})
+		return
+	}
+
+	containerComponents := devfileObj.Data.GetDevfileContainerComponents() //TODO: filter thru Console attribute, right now if there is more than one component container err out
+	if len(containerComponents) > 1 {
```

Comment:
true

Reply:
@maysunfaisal Any thoughts on the response having 0 items?

Reply:
yep we should err out when there is no component container in the devfile since that is the absolute min to get a devfile component up and running in Console.

Reply:
```suggestion
	if len(containerComponents) != 1 {
```
Is this all we need @maysunfaisal ? The error in this block is `"Console Devfile Import Dev Preview, supports only one component container"`

Reply:
Done